### PR TITLE
Update to newer `memory_addr` and `page_table_multiarch`

### PR DESCRIPTION
**Pull request by aarkegz**

### CODE REVIEW

**Review by hky1999 — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,11 @@ numeric-enum-macro = "0.2"
...
-page_table_entry = { version = "0.3.3", features = ["arm-el2"] }
-page_table_multiarch = "0.3.3"
\ No newline at end of file
+memory_addr = "0.3"
+memory_set = "0.3"
+page_table_entry = { version = "0.4", features = ["arm-el2"] }
+page_table_multiarch = "0.4"
+
+[patch.crates-io]
+page_table_entry = { git = "https://github.com/arceos-hypervisor/page_table_multiarch.git", rev = "8fd8a78" }
```
Comment (hky1999) [inline Cargo.toml:23]: Why these patches

File: src/address_space/backend/linear.rs [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
+use memory_addr::PhysAddr;
+use page_table_multiarch::{MappingFlags, PagingHandler};
 
-use crate::backend::Backend;
-use crate::npt::NestedPageTable as PageTable;
+use super::Backend;
+use crate::{npt::NestedPageTable as PageTable, GuestPhysAddr};
 
-impl Backend {
+impl<H: PagingHandler> Backend<H> {
```
Comment (hky1999) [inline src/address_space/backend/linear.rs:7]: why re-imported generic type `PagingHandler` here?
is it used for `Backend`?

**Review by hky1999 — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (aarkegz) [reply Cargo.toml]: Removed as `page_table_*` v0.4.1 released

File: src/address_space/backend/linear.rs (hunk 2 above)
Comment (aarkegz) [reply src/address_space/backend/linear.rs:7]: To implement `MappingBackend`. See `src/address_space/backend/mod.rs:54`.

File: src/addr.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+/// Guest virtual address range.
+pub type GuestVirtAddrRange = AddrRange<GuestVirtAddr>;
+/// Guest physical address range.
+pub type GuestPhysAddrRange = AddrRange<GuestPhysAddr>;
+
+#[cfg(any(target_arch = "riscv32", target_arch = "riscv64"))]
+impl page_table_multiarch::riscv::SvVirtAddr for GuestPhysAddr {
+    fn flush_tlb(_vaddr: Option<Self>) {
+        // no-op
```
[the remark this answers is not in the document]
Comment (aarkegz) [reply src/addr.rs:28]: It should be `todo!()` I think.

**Review by hky1999 — APPROVED** [review]
LGTM

**Review by hky1999 — APPROVED** [review]
LGGGGTM

**Pull request merged**